We always need to auquire the media capacity, even if it is a good old CD.

  - Rene Rebe <rene@exactcode.de>

diff -ur dvd+rw-tools-7.0/dvd+rw-mediainfo.cpp dvd+rw-tools-7.0-pathed/dvd+rw-mediainfo.cpp
--- dvd+rw-tools-7.0/dvd+rw-mediainfo.cpp	2006-09-23 17:46:57.000000000 +0000
+++ dvd+rw-tools-7.0-pathed/dvd+rw-mediainfo.cpp	2006-10-25 14:24:44.000000000 +0000
@@ -158,8 +158,7 @@
 	    }
 
 	    if (!(page2A[8+(page2A[6]<<8|page2A[7])+2]&8))
-		fprintf (stderr,":-( not a DVD unit\n"),
-		exit (FATAL_START(EINVAL));
+		{fprintf (stderr,":-( not a DVD unit\n"); goto not_dvd;}
 
 	    if (page2A==header) page2A=NULL;
 	    else if (argc>2 && !page2A_printed)
@@ -237,9 +236,8 @@
 	exit(FATAL_START(ENOMEDIUM));
     if ((profile&0xF0) != 0x10 && (profile&0xF0) != 0x20 &&
 	(profile&0xF0) != 0x40)
-	fprintf (stderr,":-( non-DVD media mounted, exiting...\n"),
-	exit (FATAL_START(EMEDIUMTYPE));
-
+	{fprintf (stderr,":-( non-DVD media mounted...\n"); goto not_dvd;}
+ 
     { const char *str;
 
 	switch (profile)
@@ -860,6 +858,8 @@
 	}
     } while (0);
 
+not_dvd:
+
     cmd[0] = 0x51;	// READ DISC INFORMATION
     cmd[8] = 32;
     cmd[9] = 0;
